modules/ppcp-settings/src/Service/AuthenticationManager.php

```diff
@@ -267,10 +286,17 @@ public function authenticate_via_oauth( bool $use_sandbox, string $shared_id, st
 		 */
 		$connection = $this->common_settings->get_merchant_data();
 
-		$connection->is_sandbox    = $use_sandbox;
-		$connection->client_id     = $credentials['client_id'];
-		$connection->client_secret = $credentials['client_secret'];
-		$connection->merchant_id   = $credentials['merchant_id'];
+		try {
+			$seller_status = $this->partners_endpoint->seller_status();
+		} catch ( PayPalApiException $exception ) {
+			$seller_status = null;
+		}
+
+		$connection->is_sandbox       = $use_sandbox;
+		$connection->client_id        = $credentials['client_id'];
+		$connection->client_secret    = $credentials['client_secret'];
+		$connection->merchant_id      = $credentials['merchant_id'];
```

Comment:
Are we sure these keys exist in `$credentials` or should we use it like `$credentials['client_id'] ?? ''`

Reply:
Re @Narek13: Those three items are **always** included in the return value of `get_credentials()`. It's a private method in the same class, so we have full control over the return value and can trust it will stay compatible.